packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -693,6 +563,66 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
       defaultProtocol: this.options.defaultProtocol,
     });
   }
+
+  private getLinkMarksInRange(
+    doc: ProsemirrorNode,
+    from: number,
+    to: number,
+    isAutoLink: boolean,
+  ): GetMarkRange[] {
+    const linkMarks: GetMarkRange[] = [];
+
+    if (from === to) {
+      const $pos = doc.resolve(from);
+      $pos.marks().forEach(() => {
+        const range = getMarkRange($pos, this.type);
+
+        if (range?.mark.attrs.auto === isAutoLink) {
+          linkMarks.push(range);
+        }
+      });
+    } else {
+      doc.nodesBetween(from, to, (node, pos) => {
+        const marks = node.marks ?? [];
+        const linkMark = marks.find(
+          ({ type, attrs }) => type === this.type && attrs.auto === isAutoLink,
+        );
+
+        if (linkMark) {
+          linkMarks.push({
+            from: pos,
+            to: pos + node.nodeSize,
+            mark: linkMark,
+            text: node.textContent,
+          });
+        }
+      });
+    }
+
+    return linkMarks;
+  }
+
+  private findAutoLinks(str: string): FoundAutoLink[] {
+    const toAutoLink: FoundAutoLink[] = [];
+    let match;
+
+    while ((match = this.options.autoLinkRegex.exec(str)) !== null) {
```

Comment:
```suggestion
    for (const match of findMatches(str, this.options.autoLinkRegex)) {
```

Maybe we can use the `findMatches` function from `@remirror/core` here. It handles the global flag and lastIndex for us.